pkg/scalers/azure_pipelines_scaler.go

```diff
@@ -139,36 +147,65 @@ type azurePipelinesMetadata struct {
 	requireAllDemands                    bool
 }
 
+type authContext struct {
+	token     string
+	expiredOn time.Time
+	pat       string
+}
+
 // NewAzurePipelinesScaler creates a new AzurePipelinesScaler
 func NewAzurePipelinesScaler(ctx context.Context, config *ScalerConfig) (Scaler, error) {
 	httpClient := kedautil.CreateHTTPClient(config.GlobalHTTPTimeout, false)
 
+	logger := InitializeLogger(config, "azure_pipelines_scaler")
 	metricType, err := GetMetricTargetType(config)
 	if err != nil {
 		return nil, fmt.Errorf("error getting scaler metric type: %w", err)
 	}
 
-	meta, err := parseAzurePipelinesMetadata(ctx, config, httpClient)
+	meta, podIdentity, err := parseAzurePipelinesMetadata(ctx, logger, config, httpClient)
 	if err != nil {
 		return nil, fmt.Errorf("error parsing azure Pipelines metadata: %w", err)
 	}
 
 	return &azurePipelinesScaler{
-		metricType: metricType,
-		metadata:   meta,
-		httpClient: httpClient,
-		logger:     InitializeLogger(config, "azure_pipelines_scaler"),
+		metricType:  metricType,
+		metadata:    meta,
+		httpClient:  httpClient,
+		podIdentity: podIdentity,
+		logger:      logger,
 	}, nil
 }
 
-func parseAzurePipelinesMetadata(ctx context.Context, config *ScalerConfig, httpClient *http.Client) (*azurePipelinesMetadata, error) {
+func getAuthMethod(config *ScalerConfig) (string, kedav1alpha1.AuthPodIdentity, error) {
+	pat := ""
+	if val, ok := config.AuthParams["personalAccessToken"]; ok && val != "" {
+		// Found the personalAccessToken in a parameter from TriggerAuthentication
+		pat = config.AuthParams["personalAccessToken"]
+	} else if val, ok := config.TriggerMetadata["personalAccessTokenFromEnv"]; ok && val != "" {
+		pat = config.ResolvedEnv[config.TriggerMetadata["personalAccessTokenFromEnv"]]
+	} else {
+		switch config.PodIdentity.Provider {
+		case "", kedav1alpha1.PodIdentityProviderNone:
+			return "", kedav1alpha1.AuthPodIdentity{}, fmt.Errorf("no personalAccessToken given or PodIdentity provider configured")
+			// return "", kedav1alpha1.AuthPodIdentity{Provider: kedav1alpha1.PodIdentityProviderNone}, nil
+		case kedav1alpha1.PodIdentityProviderAzure, kedav1alpha1.PodIdentityProviderAzureWorkload:
+			return "", kedav1alpha1.AuthPodIdentity{Provider: kedav1alpha1.PodIdentityProviderAzureWorkload}, nil
```

Comment:
`kedav1alpha1.PodIdentityProviderAzure` is deprecated, so we don't need to include it on new features, but just in case, we can include it with:
```suggestion
		case kedav1alpha1.PodIdentityProviderAzure, kedav1alpha1.PodIdentityProviderAzureWorkload:
			return "", kedav1alpha1.AuthPodIdentity{Provider: config.PodIdentity.Provider}, nil
```
The other option is removing `kedav1alpha1.PodIdentityProviderAzure`, but currently we are changing `PodIdentityProviderAzure` with `PodIdentityProviderAzureWorkload`